ui/segment.reel/segment.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @module ui/segmented-bar.reel/segment.reel
+ * @requires montage/ui/component
+ */
+var Component = require("ui/component").Component;
+
+/**
+ * @class Segment
+ * @classdesc
+ * A segment to be used inside a segmented bar.
+ * Uses FlexBox to determine rendered length.
+ *
+ * @extends Component
+ * @see SegmentedBar
+ */
+exports.Segment = Component.specialize( /** @lends Segment.prototype # */ {
+    /**
+     * @private
+     * @prop {Object} value
+     * @prop {Number} value.length
+     * @prop {String} value.label
+     */
+    _content: {value: null},
+
+    /**
+     * Length is used in draw()
+     * Label is bound to Text component in template
+     *
+     * E.g. usage to get content from SegmentedBar via binding,
+     * where `region` and `size` keys can be named anything in your data:
+     *
+     * ```json
+     * "content": {
+     *     "<-": "@segmentedBar:segment.object.{label: region, length: size}"
+     * }
+     * ```
+     *
+     * @param {Object} newContent
+     * @prop {Number} newContent.length
+     * @prop {String} newContent.label
+     * @returns {Object[]}
+     */
+    content: {
+        get: function () {return this._content},
+        set: function (newContent) {
+            if (newContent !== this._content) {
+                this._content = newContent;
+                this.needsDraw = true;
+            }
+        }
+    },
+
+    /**
+     * Redraw based on updated length
+     *
+     * @method
+     */
+    draw: {value: function () {
+        if (this.content) {
+            this.element.style.webkitBoxFlex = this.content.length;
+            this.element.style.webkitFlex = this.content.length;
+            this.element.style.msFlex = this.content.length;
+            this.element.style.flex = this.content.length;
+        }
+    }}
+});
```

Comment:
It took me about 10 minutes to figure out that "content" was a cover for a length and a label property. Why are these not separate properties? It would require two bindings instead of one, but isn't that the way object properties are usually handled? The ``content`` binding in the JSDoc is not for the faint of heart...